### PR TITLE
Crypt misc

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Some commits that are needed when using mbedtls as primary crypto library, but makes sense one their own.

### CODE REVIEW

**Comment by jforissier** [comment]
Reviewing "core: crypto: provide AES-CTS implementation":

The code is fine, but I find the commit subject slightly misleading. It is not clear that you are just moving the function wrappers from `libtomcrypt/` to `crypto/`. Maybe "core: crypto: move AES-CTS wrappers from libtomcrypt/ to crypto/"? Anyway:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/core.mk [hunk 1]
```diff
@@ -101,12 +101,19 @@ libdir = lib/libmpa
 include mk/lib.mk
 endif
 
+ifeq ($(firstword $(subst /, ,$(CFG_CRYPTOLIB_DIR))),core)
+# If a library can be compiler for both core and user space a base-prefix
```
Comment (jforissier) [inline core/core.mk]: s/compiler/compiled/

**Comment by jenswi-linaro** [comment]
Comments addressed tags applied. Note that "core: compile only libmpa with libtomcrypt" still need review.

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**